lightkurve/lightcurve.py

```diff
@@ -232,13 +232,22 @@ def flatten(self, window_length=101, polyorder=2, return_trend=False,
         # Then, apply the savgol_filter to each segment separately
         trend_signal = np.zeros(len(lc_clean.time))
         for l, h in zip(low, high):
+            # Reduce `window_length` and `polyorder` for short segments;
+            # this prevents `savgol_filter` from raising an exception
+            segment_window_length, segment_polyorder = window_length, polyorder
+            if segment_window_length > (h - l):
+                segment_window_length = h - l
+                if (segment_window_length % 2) == 0:
+                    segment_window_length -= 1  # window_length must be odd
+            if segment_polyorder >= segment_window_length:
+                segment_polyorder = segment_window_length - 1
             trend_signal[l:h] = signal.savgol_filter(x=lc_clean.flux[l:h],
-                                                     window_length=window_length,
-                                                     polyorder=polyorder, **kwargs)
+                                                     window_length=segment_window_length,
+                                                     polyorder=segment_polyorder, **kwargs)
         trend_signal = np.interp(self.time, lc_clean.time, trend_signal)
         flatten_lc = copy.deepcopy(self)
-        flatten_lc.flux /= trend_signal
-        flatten_lc.flux_err /= trend_signal
+        flatten_lc.flux = flatten_lc.flux / trend_signal
+        flatten_lc.flux_err = flatten_lc.flux_err / trend_signal
```

Comment:
Yo, I don't get the difference here xD

Reply:
The in-place operations are not the same; if you do `a += 1` it casts the result to fit back into a.  This was raising an exception when the types of `flux` and `trend_signal` were different.  For example, try this:

```
In [21]: a, b = np.array([1]), np.array([2.5])

In [22]: a /= b
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-22-b9f4626d3ca8> in <module>()
----> 1 a /= b

TypeError: ufunc 'true_divide' output (typecode 'd') could not be coerced to provided output parameter (typecode 'l') according to the casting rule ''same_kind''
```

Reply:
Oohhh, makes sense, thanks!